aws/resource_aws_kinesis_stream_consumer.go

```diff
@@ -0,0 +1,197 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAwsKinesisStreamConsumer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsKinesisStreamConsumerCreate,
+		Read:   resourceAwsKinesisStreamConsumerRead,
+		Delete: resourceAwsKinesisStreamConsumerDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(120 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				d.Set("name", d.Id())
+				return []*schema.ResourceData{d}, nil
+			},
```

Comment:
This actually needs to take both the consumer name and the stream ARN as inputs, because `DescribeStreamConsumer` requires both. Otherwise (with the code as is), importing fails with:

```
Error: Error reading Kinesis Stream Consumer: "1 validation error(s) found.", code: "InvalidParameter"
```

Following the example of `aws_route53_record` (where they take an import id of the form `<zone id>_<record name>_...`), we can do something like:

```
Importer: &schema.ResourceImporter{
	State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
		// Use @ as a delimiter because it's not allowed as part of a stream ARN or
		// a consumer name.
		// https://docs.aws.amazon.com/kinesis/latest/APIReference/API_CreateStream.html
		// https://docs.aws.amazon.com/kinesis/latest/APIReference/API_RegisterStreamConsumer.html
		parts := strings.Split(d.Id(), "@")
		if len(parts) != 2 {
			return nil, fmt.Errorf("Error importing aws_kinesis_stream_consumer. Please make sure the ID is in the form <consumer name>@<stream ARN> (e.g. my-consumer@arn:aws:kinesis:us-west-2:123456789012:stream/my-stream)")
		}

		d.Set("name", parts[0])
		d.Set("stream_arn", parts[1])
		return []*schema.ResourceData{d}, nil
	},
},
```